src/lib/array.jl

```diff
@@ -7,12 +7,17 @@
 
 @adjoint copy(x::AbstractArray) = copy(x), ȳ -> (ȳ,)
 
-_zero(xs::AbstractArray{<:Number}) = zero(xs)
-_zero(xs::AbstractArray) = Any[nothing for x in xs]
+_zero(xs::AbstractArray{<:Number}, T) = zero(xs)
+_zero(xs::AbstractArray, T) = T[nothing for x in xs]
 
 @adjoint function getindex(xs::Array, i...)
   xs[i...], function (Δ)
-    Δ′ = _zero(xs)
+    if isa(i..., NTuple{Integer})
```

Comment:
This looks like the right idea, but I guess this is meant to be `i isa NTuple{...}`.